### PR TITLE
initial diesel setup

### PR DESCRIPTION
**Pull request by izuleon**
inisiasi diesel sudah di setup untuk koneksi ke postgre yg telah ditentukan plus contoh bikin sebuah koneksi di main.rs

untuk inisiasi awal agar bisa dapat schema.rs perlu koneksi ke database asli, 
ketika di lepas si url nya akan terjadi error karena tidak ada databasenya,

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DATABASE_URL=postgres://dev:xLB3sLz0pFPnlXVB@117.54.250.100:5888/numbering_engine
```
Comment (Ujang360) [inline .env:1]: Please empty the credentials

File: .env [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+RUST_LOG=rest_api=info,actix=info,diesel_migrations=info
```
Comment (Ujang360) [inline .env:2]: Empty final newline

File: src/db.rs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+use dotenv::dotenv;
+use std::env;
+
+pub fn establish_connection() -> PgConnection {
+    dotenv().ok();
+    let database_url = env::var("DATABASE_URL")
+        .expect("DATABASE_URL must be set");
+    PgConnection::establish(&database_url)
+        .expect(&format!("Error connecting to {}", database_url))
+}
```
Comment (Ujang360) [inline src/db.rs]: Make final new line here

File: src/main.rs [hunk 4]
```diff
@@ -1,6 +1,8 @@
+mod db;
```
Comment (Ujang360) [inline src/main.rs]: Don't put this line above allocator

File: src/main.rs [hunk 5]
```diff
@@ -1,6 +1,8 @@
...
 #[global_allocator]
 static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
 
 fn main() {
     println!("Hello, world!");
+    let connection = db::establish_connection();
```
Comment (Ujang360) [inline src/main.rs]:
```suggestion
    let _ = db::establish_connection();
```

**Review by Ujang360 — CHANGES_REQUESTED** [review]
Please address comments

File: .env [hunk 6]
```diff
@@ -0,0 +1,2 @@
+DATABASE_URL=postgres://username:password@ip:port/table_name
```
Comment (yosiasm) [inline .env]:
```suggestion
DATABASE_URL=postgres://username:password@ip:port/database_name
```

**Review by Ujang360 — APPROVED** [review]
LGTM

**Review by yosiasm — CHANGES_REQUESTED** [review]
please adress comments.

File: .env (hunk 6 above)
Comment (Ujang360) [reply .env]: I prefer `numbering_engine`

File: src/schema.rs [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+joinable!(laporan -> satuan_kerja (satker_id));
+joinable!(pelapor -> satuan_kerja (satker_id));
+
+allow_tables_to_appear_in_same_query!(
+    laporan,
+    pelapor,
+    referensi,
+    satuan_kerja,
+    user,
+);
```
Comment (BimaAdi) [inline src/schema.rs:65]: How about add this file to .gitignore?. This file utomaticaly generate by diesel-cli.

**Review by BimaAdi — CHANGES_REQUESTED** [review]
add src/schema.rs to .gitignore?

File: src/schema.rs (hunk 7 above)
Comment (BimaAdi) [reply src/schema.rs:65]: *utomaticaly automaticaly

Comment (Ujang360) [reply src/schema.rs:65]: This is generated but still matters for the entire codebase

File: src/schema.rs [hunk 8]
```diff
@@ -0,0 +1,65 @@
+table! {
+    laporan (id) {
+        id -> Uuid,
+        created_date -> Timestamptz,
+        updated_date -> Nullable<Timestamptz>,
+        judul -> Varchar,
+        nomor -> Nullable<Varchar>,
+        isi -> Text,
+        status -> Laporan_status,
```
Comment (allisandrana) [inline src/schema.rs:9]: Suggestion, May need to declare the laporan_status enum type in Rust too

**Review by allisandrana — APPROVED** [review]
ok

**Review by yosiasm — APPROVED** [review]

**Review by BimaAdi — APPROVED** [review]
LGTM

File: src/schema.rs (hunk 8 above)
Comment (Ujang360) [reply src/schema.rs:9]: The `naming` should also be reconsidered, this will be tomorrow's discussion, good point @allisandrana

**Pull request merged**